### PR TITLE
Don't mark empty doctest files as skipped, fixes #1578

### LINKED ISSUE #1578: doctest marks readme files without doctests as skipped.
Here's the issue, https://travis-ci.org/graingert/dockhand/jobs/134400724#L202

on these files: https://github.com/graingert/dockhand/tree/69be7c180e67407aca462b899ff1259b3e09f080

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by The-Compiler** [comment]
I don't really understand the implications of the change, but it seems like it made a lot of tests fail :wink:

**Comment by graingert** [comment]
@The-Compiler yeah I know :p

**Comment by RonnyPfannschmidt** [comment]
i think this is related to fixture support for docktests, @nicoddemus do you remember details by chance?

File: testing/test_doctest.py [hunk 1]
```diff
@@ -14,13 +14,16 @@ def test_collect_testtextfile(self, testdir):
...
         for x in (testdir.tmpdir, checkfile):
             #print "checking that %s returns custom items" % (x,)
             items, reprec = testdir.inline_genitems(x)
             assert len(items) == 1
-            assert isinstance(items[0], DoctestTextfile)
+            assert isinstance(items[0], DoctestItem)
+            assert isinstance(items[0].parent, DoctestTextfile)
+        # Empty file has no items.
         items, reprec = testdir.inline_genitems(w)
-        assert len(items) == 1
```
Comment (graingert) [inline testing/test_doctest.py]: this looks like it was a test bug

**Comment by graingert** [comment]
requires #1582 to pass

File: testing/test_doctest.py (hunk 1 above)
Comment (nicoddemus) [inline testing/test_doctest.py]: Yep!

**Comment by nicoddemus** [comment]
Looks good, it makes much more sense for `DoctestTextfile` to collect items instead of being an item itself. :grin:

**Comment by nicoddemus** [comment]
Only missing `CHANGELOG` and `AUTHORS`, but I will update those myself before merging (right after AppVeyor passes).

**Comment by graingert** [comment]
AppVeyor is slooooooww

**Comment by nicoddemus** [comment]
Yep, it doesn't have multiple jobs like Travis, so we wait. :relieved:

**Comment by nicoddemus** [comment]
Merged, thanks again @graingert! 

(The AppVeyor failure was unrelated, will see to that in a separate PR)

**Pull request merged**